### PR TITLE
Add Google Datastore sorage handler

### PR DESCRIPTION
**Pull request by meseta**
Adds a storage backend for Google Datastore, using the `@google-cloud/datastore` NPM module.  Credentials are handled automatically by that module (by way of an GOOGLE_APPLICATION_CREDENTIALS environmental variable, a fairly standard mechanism on Google Cloud).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -198,6 +198,22 @@ Also, you must create an `uploads` table, which will store all the data for uplo
...
+
+To use the Google Datastore storage system, you must install the `@google-cloud/datastore` package via npm
+
+`npm install @google-cloud/datastore`
+
+Once you've done that, your config section should look like this:
+
+``` json
+{
+  "type": "googledatastore"
```
Comment (seejohnrun) [inline README.md]: What do you think of hyphenating this like `google-data-store.js`

File: lib/document_stores/googledatastore.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  constructor(options) {
+    this.kind = "Haste";
+    this.expire = options.expire;
+    this.datastore = new Datastore();
+  }
+
+  // Save file in a key
+  set(key, data, callback, skipExpire) {
+    var now = new Date();
+    var expireTime = (skipExpire || this.expire === undefined) ? null : new Date(now.getTime() + this.expire * 1000);
```
Comment (seejohnrun) [inline lib/document_stores/googledatastore.js]: Can use `Date.now()` here

File: README.md (hunk 1 above)
Comment (meseta) [reply README.md]: works for me, although I would suggest `google-datastore` since Google uses "datastore" as one word.

**Comment by meseta** [comment]
@seejohnrun all updated for you!

File: package.json [hunk 3]
```diff
@@ -14,6 +14,7 @@
   },
   "main": "haste",
   "dependencies": {
+    "@google-cloud/datastore": "^1.4.2",
```
Comment (seejohnrun) [inline package.json]: sorry for the delay can we remove this from the `package.json` so that it's not required for people not using this datastore?

**Pull request merged**